### PR TITLE
[Merged by Bors] - feat: bundled versions of two operations on continuous multilinear maps

### PR DESCRIPTION
**Pull request by sgouezel**
We provide versions of `smulRight` and `compContinuousLinearMap` as a continuous bilinear (resp multilinear) map.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean [hunk 1]
```diff
@@ -1199,14 +1216,49 @@ theorem compContinuousLinearMapL_apply (g : ContinuousMultilinearMap 𝕜 E₁ G
...
+For a version fixing `f` and varying `g`, see `compContinuousLinearMapL`. -/
+def compContinuousLinearMapLRight (g : ContinuousMultilinearMap 𝕜 E₁ G) :
+    ContinuousMultilinearMap 𝕜 (fun i ↦ E i →L[𝕜] E₁ i) (ContinuousMultilinearMap 𝕜 E G) :=
+  MultilinearMap.mkContinuous
+    { toFun := fun f => g.compContinuousLinearMap f
+      map_add' := by
+        intro h f i f₁ f₂
+        ext x
+        change (g fun j ↦ update f i (f₁ + f₂) j <| x j) =
+            (g fun j ↦ update f i f₁ j <| x j) + g fun j ↦ update f i f₂ j (x j)
```
Comment (rosborn) [inline Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean:1240]: My understanding is that `change` is prone to breakage if definitions change, so using the relevant theorems is more appropriate. If this is wrong, please let me know!
```suggestion
        simp only [compContinuousLinearMap_apply, add_apply]
```

File: Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean [hunk 2]
```diff
@@ -1199,14 +1216,49 @@ theorem compContinuousLinearMapL_apply (g : ContinuousMultilinearMap 𝕜 E₁ G
...
+        convert g.map_add (fun j ↦ f j (x j)) i (f₁ (x i)) (f₂ (x i)) <;>
+          exact apply_update (fun (i : ι) (f : E i →L[𝕜] E₁ i) ↦ f (x i)) f i _ _
+      map_smul' := by
+        intro h f i a f₀
+        ext x
+        change (g fun j ↦ update f i (a • f₀) j <| x j) = a • g fun j ↦ update f i f₀ j (x j)
+        convert g.map_smul (fun j ↦ f j (x j)) i a (f₀ (x i)) <;>
+          exact apply_update (fun (i : ι) (f : E i →L[𝕜] E₁ i) ↦ f (x i)) f i _ _ }
+  (‖g‖) (fun f ↦ by simp [norm_compContinuousLinearMap_le])
```
Comment (rosborn) [inline Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean:1249]:
```suggestion
    (‖g‖) (fun f ↦ by simp [norm_compContinuousLinearMap_le])
```

File: Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean [hunk 3]
```diff
@@ -930,6 +930,21 @@ theorem norm_mkPiRing (z : G) : ‖ContinuousMultilinearMap.mkPiRing 𝕜 ι z
...
+/-- Continuous bilinear map realizing `(f, z) ↦ f.smulRight z`. -/
+def smulRightL : ContinuousMultilinearMap 𝕜 E 𝕜 →L[𝕜] G →L[𝕜] ContinuousMultilinearMap 𝕜 E G :=
+  LinearMap.mkContinuous₂
+    { toFun := fun f ↦
+        { toFun := fun z ↦ f.smulRight z
+          map_add' := fun x y ↦ by ext; simp
+          map_smul' := fun c x ↦ by ext; simp [smul_smul, mul_comm] }
+      map_add' := fun f g ↦ by ext; simp [add_smul]
+      map_smul' := fun c f ↦ by ext; simp [smul_smul] }
+  1 (fun f z ↦ by simp [norm_smulRight])
```
Comment (rosborn) [inline Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean:943]:
```suggestion
    1 (fun f z ↦ by simp [norm_smulRight])
```

**Review by rosborn — COMMENTED** [review]
Looks good to me. I'm not sure mathlib's official position on when to use the `change` tactic in the library, so feel free to ignore that comment.

**Comment by sgouezel** [comment]
Thanks for the review! `change` is not really an issue when we are sure that definitions won't change (which is the case here), but still your suggestion looks better so I've committed it.

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**